### PR TITLE
B/trisolaris token list

### PR DESCRIPTION
**Pull request by memacius32**
### Do not merge until merging https://github.com/trisolaris-labs/tokens/pull/19 

- Adding Trisolaris list.
- Add support for svg logos and png as a fallback
- Replace usd tlp token symbol by usdtlp
- Commented base tokens map where not needed 
---
- Tested trisolaris list as default 
- Tested updating of token list

Screen Shot 2022-04-15 at 11 56 48
Screen Shot 2022-04-15 at 12 09 06

### CODE REVIEW

File: dev/generate_tokens/base_tokens_map.js [hunk 1]
```diff
@@ -23,174 +23,174 @@ module.exports = {
...
-      18,
-      'GBA',
-      'Golden Banana'
-    )
-  },
-
-  XNL: {
-    [ChainId.AURORA]: new Token(ChainId.AURORA, '0x7cA1C28663b76CFDe424A9494555B94846205585', 18, 'XNL', 'Chronicle')
-  }
+  // USDT: {
```
Comment (0xchain1) [inline dev/generate_tokens/base_tokens_map.js:26]: Do we not need these anymore? any reason its commented out

Comment (memacius32) [reply dev/generate_tokens/base_tokens_map.js:26]: base_tokens_map is a file for overriding the tokens that are generated when running ``yarn build tokens``, which fetches the tokens repo and builds the token constants.

 As I understand it, there is no need to override anything in the tokens repo here now

Comment (specialist0x) [reply dev/generate_tokens/base_tokens_map.js:26]: Please don't commit commented out code to production.

Comment (specialist0x) [reply dev/generate_tokens/base_tokens_map.js:26]: When building the token generator, we needed to ensure the token metadata was stable between changes.  To do this, I created `https://github.com/trisolaris-labs/interface/blob/main/src/state/stake/__tests__/stake-constants.test.ts` and `https://github.com/trisolaris-labs/interface/blob/main/src/constants/tokens/__tests__/index.test.ts`

The PR can be found here: https://github.com/trisolaris-labs/interface/pull/152

Please confirm that your new data is correct, and if it is, update the snapshot tests by running `yarn test -u`

**Review by specialist0x — CHANGES_REQUESTED** [review]
When building the token generator, we needed to ensure the token metadata was stable between changes. To do this, I created https://github.com/trisolaris-labs/interface/blob/main/src/state/stake/__tests__/stake-constants.test.ts and https://github.com/trisolaris-labs/interface/blob/main/src/constants/tokens/__tests__/index.test.ts

The PR can be found here: #152

Please confirm that your new data is correct, and if it is, update the snapshot tests by running yarn test -u

File: dev/generate_tokens/base_tokens_map.js (hunk 1 above)
Comment (memacius32) [reply dev/generate_tokens/base_tokens_map.js]: Removed🙏

**Comment by memacius32** [comment]
Updated snapshot tests

File: dev/generate_tokens/base_tokens_map.js [hunk 2]
```diff
@@ -21,176 +21,5 @@ module.exports = {
...
-      'Tether USD'
-    ),
-    [ChainId.POLYGON]: new Token(
-      ChainId.POLYGON,
-      '0xc2132d05d31c914a87c6611c10748aeb04b58e8f',
-      6,
-      'USDT',
-      'Tether USD'
-    ),
-    [ChainId.AURORA]: new Token(ChainId.AURORA, '0x4988a896b1227218e4a686fde5eabdcabd91571f', 6, 'USDT', 'Tether USD')
```
Comment (specialist0x) [inline dev/generate_tokens/base_tokens_map.js:42]: For this file, in order to allow the `tokens` repo to be the single source of truth, what do you think about only removing the `ChainID.AURORA` items from here?

That'd maintain xchain functionality if we want to do any testing or anything (as we've done in the past)

File: src/state/lists/updater.ts [hunk 3]
```diff
@@ -9,11 +9,21 @@ import { addPopup } from '../application/actions'
...
   const { library } = useActiveWeb3React()
   const dispatch = useDispatch<AppDispatch>()
   const lists = useSelector<AppState, AppState['lists']['byUrl']>(state => state.lists.byUrl)
 
+  // Code for making trisolaris list the default.
+  const lastInitializedList = useSelector<AppState, any>(state => state.lists.lastInitializedDefaultListOfLists)
+
+  if (lastInitializedList[0] === AURORA_LIST) {
+    dispatch(setDefaultList())
+  }
```
Comment (specialist0x) [inline src/state/lists/updater.ts:25]: `any` gross

```suggestion
  // Code for making trisolaris list the default.
const lastInitializedList = useSelector<AppState, AppState['lists']['lastInitializedDefaultListOfLists']>(
    state => state.lists.lastInitializedDefaultListOfLists
  )

  if (lastInitializedList?.[0] === AURORA_LIST) {
    dispatch(setDefaultList())
  }
```

**Comment by specialist0x** [comment]
image

(These should be additional PRs for our next-up/backlog), commenting here for record)

A couple issues we've had are:

1) In the upper image, due to the varying image sizes, some icon images are tiny -- is there a way to get constant size?
2) For logos that are only black and have a transparent background, they're difficult to see.  Not sure if there's a solution for this, because, in this case, a white/grey bg would fix, but if the foreground is _white_, then the bg would have to be black :|

**Comment by specialist0x** [comment]
image
Should the token lists urls be links?  I can't see the entire link, and I can't click it lol

What if we:
1. Removed the `raw.githubisercontent....`
2. Centered the list name (Trisolaris List, in this case)
3. Made `Trisolaris List` a link

wyt?

**Comment by specialist0x** [comment]
Diff looks great, @memacius32 🔥

**Comment by memacius32** [comment]
> image
> 
> (These should be additional PRs for our next-up/backlog), commenting here for record)
> 
> A couple issues we've had are:
> 
> 1. In the upper image, due to the varying image sizes, some icon images are tiny -- is there a way to get constant size?
> 2. For logos that are only black and have a transparent background, they're difficult to see.  Not sure if there's a solution for this, because, in this case, a white/grey bg would fix, but if the foreground is _white_, then the bg would have to be black :|

We can try a few things about the size in another PR.
 But about the logo colors, I think we will have to see each case and fix them or try to get another logo, at least for the tokens we are interested in.
 Fox example hapi has a yellow background logo version too:
image
 we could potentially change it (or add it to the svg)

**Comment by memacius32** [comment]
> image
> 
> Should the token lists urls be links? I can't see the entire link, and I can't click it lol
> What if we:
> 
> 1. Removed the `raw.githubisercontent....`
> 2. Centered the list name (Trisolaris List, in this case)
> 3. Made `Trisolaris List` a link
> 
> wyt?


I agree 100% on the first and second one . About the third, The arrow has a link to token lists already, but it wouldn't hurt to make the name a link to the raw json anyway

File: dev/generate_tokens/base_tokens_map.js (hunk 2 above)
Comment (memacius32) [reply dev/generate_tokens/base_tokens_map.js:42]: makes sense. Will add

**Review by specialist0x — APPROVED** [review]

**Pull request merged**